### ISSUE TITLE
pip install error, Failed building wheel for fbprophet

### ISSUE THREAD

**Issue by Alexguitar-zz** [facebook/prophet#722]
I had installed Xcode
but it's still installed fbprophet error
Does anyone can help me solve this problem?

My environment:
OSX : 10.14
python : 3.6
pip : 18.1
gcc --version :
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.44.2)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

**error messages are as below**

Building wheels for collected packages: fbprophet
  Running setup.py bdist_wheel for fbprophet ... error
  Complete output from command /Users/alexguitar/finance_mal/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-install-1e71gd5f/fbprophet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-wheel-sw_aatmh --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/fbprophet
  creating build/lib/fbprophet/stan_model
  DIAGNOSTIC(S) FROM PARSER:
  Warning: left-hand side variable (name=cp_idx) occurs on right-hand side of assignment, causing inefficient deep copy to avoid aliasing.
  Warning: left-hand side variable (name=m_pr) occurs on right-hand side of assignment, causing inefficient deep copy to avoid aliasing.
  
  INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_861b75c6337e237650a61ae58c4385ef NOW.
  error: command '/usr/bin/clang' failed with exit status 1
  
  ----------------------------------------
  **Failed building wheel for fbprophet**
  Running setup.py clean for fbprophet
Failed to build fbprophet
Installing collected packages: fbprophet
  Running setup.py install for fbprophet ... error
    Complete output from command /Users/alexguitar/finance_mal/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-install-1e71gd5f/fbprophet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-record-trpkehdh/install-record.txt --single-version-externally-managed --compile --install-headers /Users/alexguitar/finance_mal/bin/../include/site/python3.6/fbprophet:
    running install
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/fbprophet
    creating build/lib/fbprophet/stan_model
    DIAGNOSTIC(S) FROM PARSER:
    Warning: left-hand side variable (name=cp_idx) occurs on right-hand side of assignment, causing inefficient deep copy to avoid aliasing.
    Warning: left-hand side variable (name=m_pr) occurs on right-hand side of assignment, causing inefficient deep copy to avoid aliasing.
    
    INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_861b75c6337e237650a61ae58c4385ef NOW.
    error: command '/usr/bin/clang' failed with exit status 1
    
    ----------------------------------------
Command "/Users/alexguitar/finance_mal/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-install-1e71gd5f/fbprophet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-record-trpkehdh/install-record.txt --single-version-externally-managed --compile --install-headers /Users/alexguitar/finance_mal/bin/../include/site/python3.6/fbprophet" failed with error code 1 in /private/var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/pip-install-1e71gd5f/fbprophet/

**Comment by bletham** [comment]
Usually this indicates PyStan is not working. Could you check my comment in #715 and see if it is working?
Are you by chance using Anacdona?

**Comment by Alexguitar-zz** [comment]
> Usually this indicates PyStan is not working. Could you check my comment in #715 and see if it is working?
> Are you by chance using Anacdona?

Hi Bletham,
Thanks for your replying.
I used IDE(pycharm) to run your suggestion code(#715),  it will show errors as below

INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_5944b02c79788fa0db5b3a93728ca2bf NOW.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/unixccompiler.py", line 118, in _compile
    extra_postargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/ccompiler.py", line 909, in spawn
    spawn(cmd, dry_run=self.dry_run)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/spawn.py", line 36, in spawn
    _spawn_posix(cmd, search_path, dry_run=dry_run)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/spawn.py", line 159, in _spawn_posix
    % (cmd, exit_status))
distutils.errors.DistutilsExecError: command '/usr/bin/clang' failed with exit status 1
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 3, in <module>
  File "/Users/alexguitar/finance_mal/lib/python3.6/site-packages/pystan/model.py", line 350, in __init__
    build_extension.run()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/command/build_ext.py", line 339, in run
    self.build_extensions()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/command/build_ext.py", line 448, in build_extensions
    self._build_extensions_serial()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/command/build_ext.py", line 473, in _build_extensions_serial
    self.build_extension(ext)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/command/build_ext.py", line 533, in build_extension
    depends=ext.depends)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/ccompiler.py", line 574, in compile
    self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/unixccompiler.py", line 120, in _compile
    raise CompileError(msg)
distutils.errors.CompileError: command '/usr/bin/clang' failed with exit status 1

PyStan is not working, right?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
But I used Anacdona, I ran your suggestion code, it would show message as below

INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_5944b02c79788fa0db5b3a93728ca2bf NOW.
/Users/alexguitar/anaconda3/lib/python3.6/site-packages/Cython/Compiler/Main.py:367: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /var/folders/39/3y_ypsgj5m306z4g9fhg39s00000gn/T/tmpwgcverml/stanfit4anon_model_5944b02c79788fa0db5b3a93728ca2bf_8824995565144779727.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
**-0.020683351554855553**

The result is ok?
If the result is OK, do you why the reason is ?(Anacdona is ok, but pycharm is not ok)

**Comment by bletham** [comment]
That is correct, it looks like it is not working in PyCharm's python (system python?) but is working in Anaconda. Anaconda has a separate build system with its own C++ compiler, so it seems that its is working correctly while the system one is not.

I don't have enough experience debugging pystan installs in OSX to say what to do to get it working in PyCharm, but if you can use Anaconda then fbprophet should work there, since PyStan is working. Install it with conda forge:
```
conda install -c conda-forge fbprophet
```

**Comment by Alexguitar-zz** [comment]
Hi Bletham,
Thanks for your help.
I thought my problem had solved.

**Comment by saravanansaminathan** [comment]
For pip install users change the python interpreter to python 3.7 issue will get solved. With python 3.6 interpreter pystan and fbprophet problem comes

**Comment by dishantgrover** [comment]
Pip couldn't install it even with python 3.7. The same error that DLL couldn't be loaded was received.

**Comment by RSwarnkar** [comment]
I am trying to build a _docker_ container with `fbprophet` but I am getting similar error.

**Comment by RSwarnkar** [comment]
> That is correct, it looks like it is not working in PyCharm's python (system python?) but is working in Anaconda. Anaconda has a separate build system with its own C++ compiler, so it seems that its is working correctly while the system one is not.
> 
> I don't have enough experience debugging pystan installs in OSX to say what to do to get it working in PyCharm, but if you can use Anaconda then fbprophet should work there, since PyStan is working. Install it with conda forge:
> 
> ```
> conda install -c conda-forge fbprophet
> ```

Thank you so much !

**Comment by ElLorans** [comment]
I was able to install fbprophet and pystan on Windows without Anaconda following http://y-okamoto-psy1949.la.coocan.jp/Python/en1/PyStanInstalling/. I installed the components for C++ development of Desktop Applications and everything worked fine.

**Comment by garyb9** [comment]
> 
> 
> That is correct, it looks like it is not working in PyCharm's python (system python?) but is working in Anaconda. Anaconda has a separate build system with its own C++ compiler, so it seems that its is working correctly while the system one is not.
> 
> I don't have enough experience debugging pystan installs in OSX to say what to do to get it working in PyCharm, but if you can use Anaconda then fbprophet should work there, since PyStan is working. Install it with conda forge:
> 
> ```
> conda install -c conda-forge fbprophet
> ```

Thanks! works perfectly from PyCharm on Windows 10.

**Comment by praveen6993** [comment]
> conda install -c conda-forge fbprophet

thank you so much, it worked

**Comment by kanhataak** [comment]
if using Anaconda prompt install not getting error
conda install -c conda-forge fbprophet

**Comment by liaxio** [comment]
hi ! i am trying to install yfinance  plotly and fbprophet in visual studio code . but it keeps giving me errors. Can anyone help with it please ?


  ERROR: Command errored out with exit status 1:
   command: 'c:\users\sanji\onedrive\desktop\project\venv\scripts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\sanji\\AppData\\Local\\Temp\\pip-install-hozc42wx\\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\\setup.py'"'"'; __file__='"'"'C:\\Users\\sanji\\AppData\\Local\\Temp\\pip-install-hozc42wx\\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\sanji\AppData\Local\Temp\pip-wheel-mvpbhbdo'
       cwd: C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\
  Complete output (44 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib
  creating build\lib\fbprophet
  creating build\lib\fbprophet\stan_model
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\setup.py", line 122, in <module>
      setup(
    File "c:\users\sanji\onedrive\desktop\project\venv\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "c:\users\sanji\onedrive\desktop\project\venv\lib\site-packages\wheel\bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Users\sanji\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\setup.py", line 48, in run
      build_models(target_dir)
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\setup.py", line 36, in build_models
      from fbprophet.models import StanBackendEnum
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\fbprophet\__init__.py", line 8, in <module>
      from fbprophet.forecaster import Prophet
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\fbprophet\forecaster.py", line 17, in <module>
      from fbprophet.make_holidays import get_holiday_names, make_holidays_df
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\fbprophet\make_holidays.py", line 14, in <module>
      import fbprophet.hdays as hdays_part2
    File "C:\Users\sanji\AppData\Local\Temp\pip-install-hozc42wx\fbprophet_b8c1dbde35a24d22afe7dc421be9abee\fbprophet\hdays.py", line 14, in <module>
      from lunarcalendar import Lunar, Converter
  ModuleNotFoundError: No module named 'lunarcalendar'
  ----------------------------------------
  ERROR: Failed building wheel for fbprophet
  Running setup.py clean for fbprophet
  Building wheel for pystan (setup.py) ... \

**Comment by saravanansaminathan** [comment]
> lunarcalendar

Install this dependency module ** lunarcalendar**

**Comment by hiteshgupta2507** [comment]
I tried the conda package but it failed.

conda install -c conda-forge fbprophet

Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
